engine/action/sc_action.cpp

```diff
@@ -3995,14 +3998,34 @@ void action_t::do_teleport( action_state_t* state )
  */
 timespan_t action_t::calculate_dot_refresh_duration( const dot_t* dot, timespan_t triggered_duration ) const
 {
-  // WoD Pandemic
-  // New WoD Formula: Get no malus during the last 30% of the dot.
-  return std::min( triggered_duration * 0.3, dot->remains() ) + triggered_duration;
+  switch ( dot_behavior )
+  {
+    case dot_behavior_e::DOT_REFRESH_PANDEMIC:
+      return std::min( triggered_duration * 0.3, dot->remains() ) + triggered_duration;
+    case dot_behavior_e::DOT_REFRESH_DURATION:
+      return dot->time_to_next_tick() + triggered_duration;
+    case dot_behavior_e::DOT_EXTEND:
+      return dot->remains() + triggered_duration;
+    case dot_behavior_e::DOT_CLIP:
+    default:
+      return triggered_duration;
```

Comment:
Shouldn't this return timespan zero for DOT_NONE? Or is that already taken care of somewhere else?

Reply:
Default case below in L4027 will return dot_refreshable() as false for DOT_NONE so I think this just didn't breakpoint. But I think it's probably worth changing. Wouldn't be timespan::zero though in this function, would be return dot->remains() I believe?

Reply:
Ok should be fixed with https://github.com/simulationcraft/simc/pull/6191/commits/d1ffa15506c980ab74914772e9e927b5c24d3dba